src/controllers/notifications.js

```diff
@@ -70,7 +71,19 @@ const model = require('../models/notifications')
 exports.post = function * postNotification () {
   const notification = yield requestUtil.validateBody(this, 'Notification')
   log.debug('Got notification: ' + JSON.stringify(notification))
-  yield model.processNotification(notification)
-
+  try {
+    yield model.processNotification(notification)
+  } catch (e) {
+    // If this receives UnacceptableExpiryError, do not respond with HTML error 422.
+    // Doing so would cause Ledger to reply with the same message, creating an infinite loop.
+    if (e instanceof UnacceptableExpiryError) {
```

Comment:
I think the problem is that there is not a clear separation of concerns between whats in the models/payments models/notifications. The logic for checking expiry should be in models/notifications. Maybe you can add the `catch` in models/notifications for now, and it can return a value that can be used to determine the status code.